sunpy/map/mapbase.py

```diff
@@ -1965,16 +1965,40 @@ def draw_rectangle(self, bottom_left, *, width: u.deg = None, height: u.deg = No
         axes.add_artist(rect)
         return [rect]
 
-    @u.quantity_input
-    def draw_contours(self, levels: u.percent, axes=None, **contour_args):
+    def _process_levels_arg(self, levels):
+        """
+        Accept a percentage or dimensionless or map unit input for contours.
+        """
+        if not hasattr(levels, 'unit'):
+            raise TypeError("The levels argument has no unit attribute, it should be an Astropy Quantity object.")
+
+        if levels.unit.is_equivalent(u.percent):
+            return 0.01 * levels.to('percent').value * np.nanmax(self.data)
```

Comment:
I always think doing `to_value('percent')` is a bit clearer, but that's not a blocker.